tests/reporter/kubernetes.go

```diff
@@ -516,6 +527,100 @@ func (r *KubernetesReporter) logEvents(virtCli kubecli.KubevirtClient, since tim
 	fmt.Fprintln(f, string(j))
 }
 
+func (r *KubernetesReporter) dumpSRIOVInfo(virtCli kubecli.KubevirtClient) {
+	sriovOutputDir := filepath.Join(r.artifactsDir, "sriov")
+	if err := os.MkdirAll(sriovOutputDir, 0777); err != nil {
+		fmt.Fprintf(os.Stderr, "failed to create directory: %v\n", err)
+		return
+	}
+
+	r.logSRIOVNodeState(virtCli)
+	r.logSRIOVNodeNetworkPolicies(virtCli)
+	r.logNetworks(virtCli)
+	r.logOperatorConfigs(virtCli)
+
+	sriovPods, err := virtCli.CoreV1().Pods(sriovNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not list pods in sriov namespace. Error: %v", err)
+		return
+	}
+
+	events := virtCli.CoreV1().Events(sriovNamespace)
+	for _, sriovPod := range sriovPods.Items {
+		r.logSRIOVPodEvents(events, sriovPod)
+	}
+}
+
+func (r *KubernetesReporter) logSRIOVNodeState(virtCli kubecli.KubevirtClient) {
+	nodeStateLogPath := filepath.Join(r.artifactsDir, "sriov", fmt.Sprintf("%d_nodestate.log", r.failureCount))
```

Comment:
You call these methods form `dumpSRIOVInfo` where you pre-create the sriov directory. However, the dependency is not clear. To a casual user it may seem like any other reporter method which can be called individually. Would you decouple them please? Passing directory name (sriov) as a parameter or making sure that the directory exists in every method comes to mind, but I'm open for suggestions.